spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/RedisAutoConfiguration.java

```diff
@@ -91,10 +98,25 @@ public JedisConnectionFactory redisConnectionFactory()
 
 		protected final JedisConnectionFactory applyProperties(
 				JedisConnectionFactory factory) {
-			factory.setHostName(this.properties.getHost());
-			factory.setPort(this.properties.getPort());
-			if (this.properties.getPassword() != null) {
-				factory.setPassword(this.properties.getPassword());
+			if (StringUtils.hasText(this.properties.getUrl())) {
+				try {
+					URI redisURI = new URI(this.properties.getUrl());
+					factory.setHostName(redisURI.getHost());
```

Comment:
How would it make any difference as the JedisFactory didn't distinguish before? Is there any parameter I should set if it is rediss:// ?